teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -0,0 +1,149 @@
+import { useContext, useMemo, useState } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { addCriteriaToRubric } from "../transforms/addCriteriaToRubric";
+import { CatalogCriteria } from "../types/criteria";
+import { getCatalogCriteria } from "../state/helpers";
+import { ReadOnlyCriteriaDisplay } from "./ReadonlyCriteriaDisplay";
+import { Strings } from "../constants";
+import { Button } from "react-common/components/controls/Button";
+import { getReadableCriteriaTemplate, makeToast } from "../utils";
+import { setCatalogOpen } from "../transforms/setCatalogOpen";
+import { classList } from "react-common/components/util";
+import { announceToScreenReader } from "../transforms/announceToScreenReader";
+import { FocusTrap } from "react-common/components/controls/FocusTrap";
+import css from "./styling/CatalogOverlay.module.scss";
+
+interface CatalogHeaderProps {
+    onClose: () => void;
+}
+const CatalogHeader: React.FC<CatalogHeaderProps> = ({ onClose }) => {
+    return (
+        <div className={css["header"]}>
+            <span className={css["title"]}>{Strings.SelectCriteriaDescription}</span>
+            <Button
+                className={css["close-button"]}
+                rightIcon="fas fa-times-circle"
+                onClick={onClose}
+                title={Strings.Close}
+            />
+        </div>
+    );
+};
+
+interface CatalogItemLabelProps {
+    catalogCriteria: CatalogCriteria;
+    allowsMultiple: boolean;
+    existingInstanceCount: number;
+    recentlyAdded: boolean;
+}
+const CatalogItemLabel: React.FC<CatalogItemLabelProps> = ({
+    catalogCriteria,
+    allowsMultiple,
+    existingInstanceCount,
+    recentlyAdded,
+}) => {
+    const canAddMore = allowsMultiple || existingInstanceCount === 0;
+    const showRecentlyAddedIndicator = recentlyAdded && canAddMore;
+    return (
+        <div className={css["catalog-item-label"]}>
+            <div className={css["action-indicators"]}>
+                {canAddMore ? (
+                    <>
+                        <i
```

Comment:
Still related, I was thinking that the icons could also be rendered via a ternary operator. I'm not sure which would actually be faster, but I feel like it might take more time to change the css class and have the css apply to the icon to hide it rather than just saying if this is true, render this, other wise render the other one.

Reply:
The reason it's done this way is to allow us to "fade" between the two by changing opacity over time (with css transitions). If we just render one or the other conditionally, the immediate switch can look a little jarring.